deployment.rb

```diff
@@ -285,6 +285,20 @@ def method_missing(*args)
   def with_default(default_value)
     DelegateWithDefault.new(self, default_value)
   end
+
+  # When running on Chef Server, use Search via knife CLI to fetch a dynamic list of app-server front-ends,
+  # appending to the static list already provided by configuration files.
+  def app_servers
+    return super unless CDO.chef_managed
+    require 'cdo/rake_utils'
+    servers = RakeUtils.with_bundle_dir(cookbooks_dir) do
+      knife_cmd = "knife search node 'roles:front-end AND chef_environment:#{rack_env}' --format json --attribute cloud_v2"
```

Comment:
Could you refactors this up as a script that developers can run themselves to see the current list of production front-ends independently of deployment.  I think developers might otherwise find it hard to know exactly what the frontends are now.